bin/skyfactory/merge_buzzard.py

```diff
@@ -330,11 +329,11 @@ def process_single_file(self, obs, truth, pz, sample, rank, debug=False):
             ifile = 0
 
         gout = fio.FITS(self.odir + '/' + self.simname +
-                        '_{}'.format(self.obsdir[:-1]) + '_gold.{}.fits'.format(rank), 'rw')
+                        '_{}'.format(self.obsname[:-1]) + '_gold.{}.fits'.format(rank), 'rw')
         sout = fio.FITS(self.odir + '/' + self.simname +
-                        '_{}'.format(self.obsdir[:-1]) + '_shape.{}.fits'.format(rank), 'rw')
+                        '_{}'.format(self.obsname[:-1]) + '_shape.{}.fits'.format(rank), 'rw')
         pout = fio.FITS(self.odir + '/' + self.simname +
-                        '_{}'.format(self.obsdir[:-1]) + '_pz.{}.fits'.format(rank), 'rw')
+                        '_{}'.format(self.obsname[:-1]) + '_pz.{}.fits'.format(rank), 'rw')
 #        alout = fio.FITS(self.odir + '/' + self.simname +
```

Comment:
In this way, we can preserve the original feature but also have more flexibility on setting obsdir.